Samples/MapScenarios/MapScenarios/Scenarios/WalkTimeViewController.swift

```diff
@@ -23,51 +23,139 @@ class WalkTimeViewController: UIViewController {
     let locationManager = CLLocationManager()
     var firstLocationAcquired = false
     
+    // MARK: Walk Time
+    // Walk speed is in meters/second
+    var walkSpeed: Double {
+        if let calculatedWalkSpeed = calculatedWalkSpeed, acceptedWalkSpeedRange.contains(calculatedWalkSpeed) {
+            return calculatedWalkSpeed
+        }
+        return defaultWalkSpeed
+    }
+    let defaultWalkSpeed = 1.4
+    var calculatedWalkSpeed: Double?
+    let acceptedWalkSpeedRange = 0.5..<5.0
+    var estimatedWalkTime: Double?
+    var lastLocationUpdates = [CLLocation]()
+    let walkTimeLabel = UILabel()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
         navigationItem.title = "Walk Time Calculation"
         
-        if applicationId.count > 0 && accessKey.count > 0 && signatureKey.count > 0 && buildingIdentifier != 0 {
-            PWCore.setApplicationID(applicationId, accessKey: accessKey, signatureKey: signatureKey)
-        } else {
-            fatalError("applicationId, accessKey, signatureKey, and buildingIdentifier must be set")
+        if !validateBuildingSetting(appId: applicationId, accessKey: accessKey, signatureKey: signatureKey, buildingId: buildingIdentifier) {
+            return
         }
         
-        locationManager.delegate = self
-        locationManager.requestWhenInUseAuthorization()
-        
+        PWCore.setApplicationID(applicationId, accessKey: accessKey, signatureKey: signatureKey)
```

Comment:
I see this code a lot. Can we refactor into a base view controller that we just reuse?